server/src/main/java/org/elasticsearch/action/admin/indices/create/AutoCreateAction.java

```diff
@@ -0,0 +1,244 @@
+        }
+
+        private static List<String> getExceptionMessages(final Map<String, Exception> map) {
+            return map.values().stream().map(Throwable::getMessage).sorted(String::compareTo).collect(Collectors.toList());
+        }
+
+        private static List<String> getKeys(final Map<String, Exception> map) {
+            return map.keySet().stream().sorted(String::compareTo).collect(Collectors.toList());
+        }
+    }
+
+    public static final class TransportAction extends TransportMasterNodeAction<Request, Response> {
+
+        private final Client client;
+
+        @Inject
+        public TransportAction(TransportService transportService, ClusterService clusterService, ThreadPool threadPool,
+                               ActionFilters actionFilters, IndexNameExpressionResolver indexNameExpressionResolver, Client client) {
+            super(NAME, transportService, clusterService, threadPool, actionFilters, Request::new, indexNameExpressionResolver);
+            this.client = client;
+        }
+
+        @Override
+        protected String executor() {
+            return ThreadPool.Names.SAME;
+        }
+
+        @Override
+        protected Response read(StreamInput in) throws IOException {
+            return new Response(in);
+        }
+
+        @Override
+        protected void masterOperation(Task task,
+                                       Request request,
+                                       ClusterState state,
+                                       ActionListener<Response> listener) {
+            autoCreate(request, listener, client);
+        }
+
+        @Override
+        protected ClusterBlockException checkBlock(Request request, ClusterState state) {
+            return state.blocks().indicesBlockedException(ClusterBlockLevel.METADATA_WRITE, request.names.toArray(new String[0]));
+        }
+    }
+
+    static void autoCreate(Request request, ActionListener<Response> listener, Client client) {
+        // For now always redirect to the auto create index action, because only indices get auto created.
+        final AtomicInteger counter = new AtomicInteger(request.getNames().size());
+        final Map<String, Exception> results = new HashMap<>();
+        for (String name : request.getNames()) {
+            CreateIndexRequest createIndexRequest = new CreateIndexRequest();
+            createIndexRequest.index(name);
+            createIndexRequest.cause(request.getCause());
+            createIndexRequest.masterNodeTimeout(request.masterNodeTimeout());
+            createIndexRequest.preferV2Templates(request.getPreferV2Templates());
+            client.execute(AutoCreateIndexAction.INSTANCE, createIndexRequest, ActionListener.wrap(
+                createIndexResponse -> {
+                    // Maybe a bit overkill to ensure visibility of results map across threads...
+                    synchronized (results) {
+                        results.put(name, null);
```

Comment:
The map is in the response named "failureByNames". I think we should either not add successful indices into it, rename the map or separate the two parts in the response.